gnocchi/tests/test_aggregates.py

```diff
@@ -1082,3 +1083,265 @@ def test_add_and_get_measures_with_holes(self):
+        ])
+        self.trigger_processing([str(self.metric.id), str(metric2.id)])
+
+        values = processor.get_measures(
+            self.storage, [(self.metric, "mean"), (metric2, "mean")],
+            ["-", ["resample", "mean", numpy.timedelta64(1, 'D'),
+                   ["metric",
+                    [str(self.metric.id), "mean"],
+                    [str(metric2.id), "mean"]]], 2],
+            granularity=numpy.timedelta64(1, 'h'))
+
+        self.assertEqual({
+            "%s_%s" % (self.metric.id, "mean"): [
+                (datetime64(2014, 1, 1, 0, 0, 0),
+                 numpy.timedelta64(1, 'D'), 37.75)],
+            "%s_%s" % (metric2.id, "mean"): [
+                (datetime64(2014, 1, 1, 0, 0, 0),
+                 numpy.timedelta64(1, 'D'), 2.75)],
+        }, values)
+
+    def test_rolling(self):
+        metric2, __ = self._create_metric()
+        self.incoming.add_measures(self.metric, [
+            incoming.Measure(datetime64(2014, 1, 1, 12, 0, 1), 69),
+            incoming.Measure(datetime64(2014, 1, 1, 12, 5, 31), 42),
+            incoming.Measure(datetime64(2014, 1, 1, 12, 10, 31), 4),
+            incoming.Measure(datetime64(2014, 1, 1, 12, 15, 45), 44),
+        ])
+        self.incoming.add_measures(metric2, [
+            incoming.Measure(datetime64(2014, 1, 1, 12, 0, 5), 9),
+            incoming.Measure(datetime64(2014, 1, 1, 12, 5, 41), 2),
+            incoming.Measure(datetime64(2014, 1, 1, 12, 10, 31), 4),
+            incoming.Measure(datetime64(2014, 1, 1, 12, 15, 10), 4),
+        ])
+        self.trigger_processing([str(self.metric.id), str(metric2.id)])
+
+        values = processor.get_measures(
+            self.storage, [(self.metric, "mean"), (metric2, "mean")],
+            ["/", ["rolling", "sum", 2,
+                   ["metric", [str(self.metric.id), "mean"],
+                    [str(metric2.id), "mean"]]], 2],
+            granularity=numpy.timedelta64(5, 'm'))
+
+        self.assertEqual({
+            "%s_%s" % (self.metric.id, "mean"): [
+                (datetime64(2014, 1, 1, 12, 5, 0),
+                 numpy.timedelta64(5, 'm'), 55.5),
+                (datetime64(2014, 1, 1, 12, 10, 0),
+                 numpy.timedelta64(5, 'm'), 23),
+                (datetime64(2014, 1, 1, 12, 15, 0),
+                 numpy.timedelta64(5, 'm'), 24)
+            ],
+            "%s_%s" % (metric2.id, "mean"): [
+                (datetime64(2014, 1, 1, 12, 5, 0),
+                 numpy.timedelta64(5, 'm'), 5.5),
+                (datetime64(2014, 1, 1, 12, 10, 0),
+                 numpy.timedelta64(5, 'm'), 3),
+                (datetime64(2014, 1, 1, 12, 15, 0),
+                 numpy.timedelta64(5, 'm'), 4),
+            ],
+        }, values)
+
+        # By references
+        values = processor.get_measures(
+            self.storage, [(self.metric, "mean"), (metric2, "mean")],
+            ["/", ["rolling", "sum", 2, [
```

Comment:
what am i missing here, it looks like exactly same operations as above

Reply:
Indeed, removed. That's scories from an old patchset.